pandas/tests/series/test_constructors.py

```diff
@@ -251,6 +251,20 @@ def test_constructor_sanitize(self):
         s = Series(np.array([1., 1., np.nan]), copy=True, dtype='i8')
         self.assertEqual(s.dtype, np.dtype('f8'))
 
+    def test_constructor_copy(self):
+        # GH15125
+        # test dtype parameter has no side effects on copy=True
+        x = Series(np.array([1.]))
```

Comment:
do this in a loop, testing the for both of these as input to the Series.

``[1.], np.array([1.])``